ambry-network/src/main/java/com/github/ambry/network/NettyServerRequestResponseChannel.java

```diff
@@ -100,57 +100,14 @@ public void closeConnection(NetworkRequest originalRequest) throws InterruptedEx
 
   @Override
   public NetworkRequest receiveRequest() throws InterruptedException {
-    NetworkRequest request;
-    while (true) {
-      request = networkRequestQueue.take();
-      http2ServerMetrics.requestQueuingTime.update(System.currentTimeMillis() - request.getStartTimeInMs());
-      if (consumeSizeHeader(request)) {
-        return request;
-      }
-    }
+    NetworkRequest request = networkRequestQueue.take();
```

Comment:
I actually removed the while loop deliberately. Earlier, it was possible that reading the first eight bytes of a dequeued request could fail. Due to that, we had to loop until we found a valid request. Now, since the reading the first eight bytes happens in `AmbryNetworkRequestHandler.java`, the dequeued request is always valid. Hence, we don't need while loop. Please let me know if I missed something here.